risc0/circuit/recursion/src/prove/witgen.rs

```diff
@@ -98,6 +98,23 @@ where
             )
             .context("witness generation failure")?;
 
+        // Add random noise to end of the data columns
+        scope!("noise", {
+            use risc0_zkp::ZK_CYCLES;
+            let mut rng = rand::rng();
+            let noise = vec![BabyBearElem::random(&mut rng); ZK_CYCLES * CIRCUIT.data_size()];
+            hal.eltwise_copy_elem_slice(
+                &data,
+                &noise,
+                CIRCUIT.data_size(),      // from_rows
+                ZK_CYCLES,                // from_cols
+                0,                        // from_offset
+                ZK_CYCLES,                // from_stride
+                total_cycles - ZK_CYCLES, // into_offset
+                total_cycles,             // into_stride
+            );
+        });
+
```

Comment:
Does this imply that noise was not being added? I'm a bit confused by this.

Reply:
That's right. When Jeremy and I took a look at this, in debugging the issue he noticed that this was missing.